charts/payment-integration/templates/secrets.yaml

```diff
@@ -0,0 +1,17 @@
+## The secret resource is required in order to hide secrets within the K8s UI. If we skip the usage of secrets yaml,
+## sensitive information would be shown in plain text i.e. in the Google console.
+apiVersion: v1
+kind: Secret
+metadata:
+  name: {{ template "payment.name" . }}-secret
```

Comment:
naming should be consistent across all k8s resources of this chart. either use template `{{ template "payment.name" . }}`or `{{ template "payment.fullname" . }}` (see `deployment.yaml`, `ingress.yaml`, `service.yaml`etc. )